src/components/SearchBar.js

```diff
@@ -105,22 +109,33 @@ export default function SearchBar(props) {
       });
     }
 
+    const handleToggleTemperature = (e) => {
+        setFilterTemperature( filterTemperature => {
+            const result = !filterTemperature;
+            if (result) handlePerformFilter("temperature")
+            else handlePerformFilter(false)
+            return result;
+        })
+    }
+
     const SearchStartAdornment = () => (
       <InputAdornment position="start">
         <Tooltip title="Search"><IconButton onClick={handlePerform} disabled={query === ""} size="small"><SearchIcon /></IconButton></Tooltip>
       </InputAdornment>
     )
 
-    const SearchEndAdornment = () => (
-      <InputAdornment position="end">
-        {count ? <span className={classes.result}>{pos} / {count}</span> : (null)}
-        <Tooltip title="Toggle regular expression mode"><IconButton onClick={handleToggleRegex} color={regexMode ? 'primary' : 'default'} size="small"><RegexIcon /></IconButton></Tooltip>
-        <Tooltip title="Toggle case sensitivity"><IconButton onClick={handleToggleCaseSensitive} color={caseSensitive ? 'primary' : 'default'} size="small"><CaseIcon /></IconButton></Tooltip>
-        <Tooltip title="Scroll to previous match"><IconButton onClick={props.onPrev} disabled={count === 0} size="small"><PrevIcon /></IconButton></Tooltip>
-        <Tooltip title="Scroll to next match"><IconButton onClick={props.onNext} disabled={count === 0} size="small"><NextIcon /></IconButton></Tooltip>
-        <Tooltip title="Delete search query"><IconButton onClick={handleCancel} disabled={query === ""} size="small"><CancelIcon /></IconButton></Tooltip>
-      </InputAdornment>
-    )
+    const SearchEndAdornment = (props) => {
+        return(
+            <InputAdornment position="end">
+              {count ? <span className={classes.result}>{pos} / {count}</span> : (null)}
+              {props.showTemperatureFilter ? <Tooltip title="Toggle filter temperature"><IconButton onClick={handleToggleTemperature} color={filterTemperature ? 'primary' : 'default'} size="small"><FilterIcon /></IconButton></Tooltip> : (null)}
+              <Tooltip title="Toggle regular expression mode"><IconButton onClick={handleToggleRegex} color={regexMode ? 'primary' : 'default'} size="small"><RegexIcon /></IconButton></Tooltip>
+              <Tooltip title="Toggle case sensitivity"><IconButton onClick={handleToggleCaseSensitive} color={caseSensitive ? 'primary' : 'default'} size="small"><CaseIcon /></IconButton></Tooltip>
+              <Tooltip title="Scroll to previous match"><IconButton onClick={props.onPrev} disabled={count === 0} size="small"><PrevIcon /></IconButton></Tooltip>
+              <Tooltip title="Scroll to next match"><IconButton onClick={props.onNext} disabled={count === 0} size="small"><NextIcon /></IconButton></Tooltip>
+              <Tooltip title="Delete search query"><IconButton onClick={handleCancel} disabled={query === ""} size="small"><CancelIcon /></IconButton></Tooltip>
+            </InputAdornment>)
+    }
```

Comment:
Why the switch to a `{ return ... }` here? 🤔

And this breaks utilization of the outer `props.onPrev`, `props.onNext` and thus the search function.